src/app/collective-rewards/rewards/builders/hooks/useGetBuildersRewards.ts

```diff
@@ -30,28 +32,39 @@ export type BuildersRewards = {
 }
 
 export const useGetBuildersRewards = ({ rif, rbtc }: { [token: string]: Token }, currency = 'USD') => {
-  const { data: builders, isLoading: searchLoading, error: searchError } = useGetBuildersByState()
-  const buildersV2 = builders as Required<Builder>[]
+  const {
+    initialState: { allocations },
+    state: { getBuilderIndexByAddress },
+  } = useContext(AllocationsContext)
+  let {
+    data: builders,
+    isLoading: searchLoading,
+    error: searchError,
+  } = useGetBuildersByState<Required<Builder>>()
+
+  const filteredBuilders = builders.filter(({ address, stateFlags }) => {
+    if (!stateFlags.kycApproved || stateFlags.revoked) {
+      const allocation = allocations[getBuilderIndexByAddress(address)]
+      if (!allocation || allocation === 0n) return false
+    }
+
+    return true
+  })
+
   const {
     data: totalPotentialRewards,
     isLoading: totalPotentialRewardsLoading,
     error: totalPotentialRewardsError,
   } = useGetTotalPotentialReward()
 
-  const buildersAddress = buildersV2.map(({ address }) => address)
+  const buildersAddress = builders.map(({ address }) => address)
   const {
     data: buildersRewardsPct,
     isLoading: buildersRewardsPctLoading,
     error: buildersRewardsPctError,
   } = useGetBuildersRewardPercentage(buildersAddress)
 
-  // TODO: validate what do to here
-  // Leaderboard only shows active builders or inactive builders that the backer allocated to
-  /* const activeOrAllocatedBuilders = activeBuilders?.filter(
-      (builder, i) => builder.status === 'Active' || (allBackerAllocations && allBackerAllocations?.[i] > 0n),
-    ) */
-
-  const gauges = buildersV2.map(({ gauge }) => gauge)
+  const gauges = builders.map(({ gauge }) => gauge)
```

Comment:
Should we use `filteredBuilders` here?

Reply:
@franciscotobar here I think it's fine that we use the builders because we look for the total allocations of all the builders 👍🏼